python/lsst/ts/ataos/ataos_csc.py

```diff
@@ -782,78 +993,92 @@ async def correction_loop(self):
         """Coroutine to send corrections to m1, m2, hexapod and focus at
         the heartbeat frequency."""
 
+        # Declare an asyncio lock so no offsets can be added while
+        # the corrections are being applied
+        correction_loop_lock = asyncio.Lock()
+        # Create an Event object that will get published after each
+        # loop iteration before the sleep.
+
+        self.correction_loop_completed_evt = asyncio.Event()
```

Comment:
Moved into __init__. 
The event saying when a correction_loop iteration has been completed is still required because we don't want to release commands (e.g. offset of some sort) until we know the offsets have been applied. So this means we need a way to know that the correction_loop task has run once before returning, which is done by listening for the event.